test/test_mpi_p4est_2d.jl

```diff
@@ -149,6 +149,66 @@ const EXAMPLES_DIR = pkgdir(Trixi, "examples", "p4est_2d_dgsem")
             @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
         end
     end
+
+    # Test MPI-parallel handling of .inp meshes generated by HOHQMesh
+    @trixi_testset "elixir_euler_wall_bc_amr.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_euler_wall_bc_amr.jl"),
+                            l2=[
+                                0.02026685991647352,
+                                0.017467584076280237,
+                                0.011378371604813321,
+                                0.05138942558296091
+                            ],
+                            linf=[
+                                0.35924402060711524,
+                                0.32068389566068806,
+                                0.2361141752119986,
+                                0.9289840057748628
+                            ],
+                            tspan=(0.0, 0.15))
+        # Ensure that we do not have excessive memory allocations
+        # (e.g., from type instabilities)
+        let
+            t = sol.t[end]
+            u_ode = sol.u[end]
+            du_ode = similar(u_ode)
+            @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 1000
+        end
+    end
```

Comment:
This test is unrelated to this PR. But HOHQMesh generated meshes + nontrivial (i.e., non `:all`) boundary conditions are not tested in 2D (but in 3D):

https://github.com/trixi-framework/Trixi.jl/blob/dc8cd7e4489e605f8a122ac0bff0d687faa6e20e/test/test_mpi_p4est_3d.jl#L197-L223